sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -809,7 +805,12 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
     }
 
     if (!path.isEmpty) {
-      val customLocation = Some(CatalogUtils.stringToURI(path))
+      val customLocation = if (ctx.LOCAL() != null) {
+        // force scheme to be file rather than fs.default.name
+        Some(UriBuilder.fromUri(CatalogUtils.stringToURI(path)).scheme("file").build())
```

Comment:
Hey, I think you gotta take a look for https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala#L147